Tensile/KernelWriterSource.py

```diff
@@ -833,8 +839,11 @@ def functionSignature(self, kernel ):
     if self.language == "HIP":
       #s += "  hipLaunchParm lp," + self.endLine
       globalStr = ""
-    ptrStr = kernel["ProblemType"]["DestDataType"].toDevice(self.language) \
-        if not kernel["_GlobalAccumulation"] else "float"
+    ptrStr = kernel["ProblemType"]["DestDataType"].toDevice(self.language)
+    if kernel["_GlobalAccumulation"]:
+      ptrStr = kernel["ProblemType"]["ComputeDataType"].toDevice(self.language)
+      if kernel["ProblemType"]["DataType"].isHalf() and kernel["ProblemType"]["HighPrecisionAccumulate"]:
+        ptrStr = "float"
```

Comment:
having this string here seems a little fragile - is there a way to instantiate a float datatype and then use the toDevice(self.language) on it?

Reply:
modify to use DataType.